source/common/router/upstream_request.h

```diff
@@ -195,6 +195,7 @@ class UpstreamRequest : public Logger::Loggable<Logger::Id::router>,
   std::unique_ptr<GenericUpstream> upstream_;
   absl::optional<Http::StreamResetReason> deferred_reset_reason_;
   Upstream::HostDescriptionConstSharedPtr upstream_host_;
+  std::shared_ptr<StreamInfo::UpstreamInfoImpl> upstream_info_;
```

Comment:
if we're going to eagerly create this, I think we can nix the local upsteam_host_ to not have it stored in many places.